src/serialization/src/main/kotlin/org/vaccineimpact/api/serialization/ModelBinder.kt

```diff
@@ -3,29 +3,30 @@ package org.vaccineimpact.api.serialization
 import org.vaccineimpact.api.models.ErrorInfo
 import org.vaccineimpact.api.models.validation.CanBeBlank
 import org.vaccineimpact.api.serialization.validation.ValidationException
-import org.vaccineimpact.api.serialization.validation.applyRule
-import org.vaccineimpact.api.serialization.validation.checkMissing
+import org.vaccineimpact.api.serialization.validation.Validator
 import kotlin.reflect.KClass
 import kotlin.reflect.KProperty1
 import kotlin.reflect.full.findAnnotation
 import kotlin.reflect.full.memberProperties
 
-class ModelBinder(private val serializer: Serializer = MontaguSerializer.instance)
+class ModelBinder(private val serializer: Serializer = MontaguSerializer.instance,
+                  private val validator: Validator = Validator(serializer))
 {
     fun <T : Any> deserialize(body: String, klass: Class<T>): T
     {
         val model = serializer.fromJson(preprocessed(body), klass)
         val errors = verify(model)
         if (errors.any())
         {
-            throw ValidationException(errors)
+            throw ValidationException(errors.distinct())
```

Comment:
A `RequiredWhen` check may return a validation error for a null dependent property, which will also be returned by the regular null check, so there could be duplicates.